app/controllers/posts_controller.rb

```diff
@@ -1,14 +1,45 @@
 class PostsController < ApplicationController
-  # index action displaying a list of posts belonging to a specific user
-  # first retrieves the user record base on user_id
-  # next fetches all posts associated with that user using @user.posts
+  before_action :set_user, only: %i[index create]
+  before_action :set_post, only: %i[show like]
 
   def index
-    @user = User.find(params[:user_id])
     @posts = @user.posts
   end
 
   def show
+    @post = Post.find(params[:id]) # Find the post by its id
+    @comment = Comment.new
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = @user.posts.build(post_params)
+    if @post.save
+      redirect_to user_post_path(@user, @post), notice: 'Post created successfully.'
+    else
+      render :new
+    end
+  end
+
+  def like
+    @post.likes.create(user: current_user)
+    redirect_to post_path(@post), notice: 'Post liked successfully.'
+  end
+
```

Comment:
- [ ] Creation of like should not be implemented in the Posts controller. Like should have its own controller with its own `create` action. Kindly add these and remove like feature from the Posts controller.